File: services/typescript/koa/src/routes/middleware/invalid_method.ts
import type { Middleware } from "koa";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "telemetry";

export const invalidMethodMiddleware: Middleware = (ctx) => {
  getLogger().warn(
    { method: ctx.req.method, path: ctx.path },
    "Invalid method"
  );

  sendError(ctx, AppErrors.InvalidMethod);
};

File: services/go/gin/routes/handlers/invalid_method.go
package handlers

import (
	"gin/routes"
	"log/slog"

	"github.com/gin-gonic/gin"
)

func InvalidMethodHandler(c *gin.Context) {
	slog.Warn("Invalid method", "method", c.Request.Method, "path", c.Request.URL.Path)

	routes.SendError(c, routes.AppErrors.InvalidMethod)
}
